### ISSUE TITLE
batch norm in tch-rs

### ISSUE THREAD

**Issue by zeroexcuses** [LaurentMazare/tch-rs#84]
I can run the following code in python:

```
import torch
import torch.nn as nn

x = torch.randn(10, 40)
bn1 = nn.BatchNorm1d(num_features = 40)
y = bn1(x);
```

However, when I try to run:

```
#[test]
fn my_test() {
    let opts = (tch::Kind::Float, tch::Device::Cpu);
    let vs = nn::VarStore::new(tch::Device::Cpu);

    let x = Tensor::randn(&[10, 40], opts);
    let bn1 = nn::batch_norm1d(vs.root(), 40, nn::BatchNormConfig::default());
    let y = bn1.forward_t(&x, true);
    //    main(tch::Kind::Float, tch::Devi ce::Cuda(0))
}
```

I get an error of:

```
panicked at 'expected an input tensor with 3 dims, got [10, 40]', 
```

My confusion here is that I don't see why BatchNorm can't be run on a 2D dataset, where

num_rows = N = total number of samples in this batch
num_cols = feature input dimension

**Comment by LaurentMazare** [comment]
It feels a bit to me that this should be called `BatchNorm0d` rather than `BatchNorm1d` in this case as `BatchNorm2d` takes as input 4d tensors and `BatchNorm3d` 5d tensors ([pytorch doc](https://pytorch.org/docs/stable/nn.html#batchnorm3d)).
Anyway `BatchNorm1d` in python indeed allows for 2d inputs so I've tweaked the rust bindings to do the same.

**Comment by zeroexcuses** [comment]
Thanks!

I agree, it sounds like the old way of naming is more consistent.

Under the old code, is

```
    let x = Tensor::randn(&[10, 40], opts);
    let bn1 = nn::batch_norm1d(vs.root(), 40, nn::BatchNormConfig::default());
    let y = bn1.forward_t(&x.view(&[10, 40, 1]), true).view(&[10, 40]);
 ```

correct? From what I read, in the "3d" case, it's treated as (N, C, L), and the vars are normalized over C for every (N, *, L).

**Comment by LaurentMazare** [comment]
I think your code is correct, the [python docs](https://pytorch.org/docs/stable/nn.html#torch.nn.BatchNorm1d) indeed mention NCL.
The old naming felt a bit better but I would prefer keeping things as close as possible to the python api so that it's easier to port over models/tutorials/...

**Comment by zeroexcuses** [comment]
Thanks for the guidance!
